src/math/Vector3.js

```diff
@@ -695,6 +696,12 @@ Object.assign( Vector3.prototype, {
 
 	},
 
+	toVector2: function () {
+
+		return new Vector2( this.x, this.y );
```

Comment:
Sorry, but we avoid this pattern. In three.js it is the user's responsibility to instantiate objects. (The `clone()` method is an exception.)

Reply:
A pattern we could support -- in theory at least -- is something like:
```js
vector3.setFromVector2( vector2, z );

vector2.setFromVector3( vector3 );
```
but I do not see a compelling reason for those methods.

The `.set()` method is sufficient -- and also more flexible.

Reply:
This is sad, as this change can make quite a lot of code shorter. Example:

```javascript
// current way
const direction1Vec2 = myObject1.getDirection();
const direction2Vec2 = myObject2.getDirection();
const angleDirection = new Vector3(direction1Vec2.x, direction1Vec2.y).angleTo(new Vector3(direction2Vec2.x, direction2Vec2.y));
// with this change
const angleDirection = myObject1.getDirection().toVector3().angleTo(myObject2.getDirection().toVector3());
```

I personally think that such conversion methods are a similarly justified case as the `clone` methods are. But If you do not see it this way, I guess we will have to resort to our own converter helper functions.

Reply:
Is that code called every frame in your app? If so, create working `Vector3`'s and reuse them to avoid unnecessary garbage collection.

This is why we avoid adding methods to the library similar to the ones you proposed.